### PR TITLE
CI/CD: both enhance commit skeleton test in ubuntu18.04 and centos 8.2 of SGX2

### PR DESCRIPTION
**Pull request by hustliyilin**
Split commit skeleton test into two containers:
- rune_dev container: simulate development environment, used to make
  inclavare containers rpm or deb packages and build the app images needed for testing.
- test container: based on ubuntu18.04 or centos 8.2, used to install
  inclavare containers rpm or deb packages and test the app images.

Signed-off-by: Yilin Li YiLin.Li@linux.alibaba.com

### CODE REVIEW

File: .github/workflows/commit_skeleton_ex.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      with:
+        ref: refs/pull/${{ github.event.pull_request.number }}/merge
+
+    - name: Prepare work
+      run: echo "CPU_NUM=$(nproc --all)" >> $GITHUB_ENV;
+        echo "RUNE_VERSION=$(grep 'Version:' rune/dist/rpm/rune.spec | awk '{print $2}')" >> $GITHUB_ENV;
+
+    - name: Create dev container
+      run: |
+        rune_dev=$(docker run -itd --privileged --rm --net host --device /dev/sgx_enclave --device /dev/sgx_provision -v $GITHUB_WORKSPACE:/root/inclavare-containers rune-dev:${{ matrix.tag }});
```
Comment (jiazhang0) [inline .github/workflows/commit_skeleton_ex.yml:27]: The device passthrough is wrong.

Comment (hustliyilin) [reply .github/workflows/commit_skeleton_ex.yml]: > The device passthrough is wrong.

Done, thanks.

File: .github/workflows/commit_skeleton_ex.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    - name: Install ubuntu dependency
+      if: ${{ contains(matrix.tag, 'ubuntu') }}
+      run: docker exec $inclavare_test bash -c 'apt-get update -y && apt-get install -y wget iptables gnupg libprotobuf-c1 libbinutils;
+        echo "deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main" | tee /etc/apt/sources.list.d/intel-sgx.list && wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add -; 
+        apt-get update -y && apt-get install -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl;
+        cd /root/inclavare-containers;
+        dpkg -i rune*.deb;
+        dpkg -i shim*.deb;
+        dpkg -i sgx-tools*deb'
```
Comment (jiazhang0) [inline .github/workflows/commit_skeleton_ex.yml:110]: *.deb

File: .github/workflows/commit_skeleton_ex.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        
+    - name: Start docker daemon
+      run: |
+        docker exec $inclavare_test bash -c "dockerd -b docker0 --storage-driver=vfs &"
+
+    - name: Load docker image
+      run: |
+        docker exec $inclavare_test bash -c "cd /root/inclavare-containers//${{ matrix.tag }};
+        docker load -i skeleton-enclave.tar;
+        rm skeleton-enclave.tar"
```
Comment (jiazhang0) [inline .github/workflows/commit_skeleton_ex.yml:133]: use -f

File: .github/workflows/commit_skeleton_ex.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    - name: Run skeleton v2 bundle
+      if: always()
+      run: |
+        docker exec $inclavare_test bash -c "mkdir rootfs;
+        docker create --name skeleton-enclave skeleton-enclave;
+        docker export skeleton-enclave | tar -C rootfs -xvf -;
+        cp /etc/resolv.conf rootfs/etc/resolv.conf;
+        mkdir -p /var/run/rune;
+        cp /usr/lib/liberpal-skeleton-v2.so /var/run/rune/liberpal-skeleton-v2.so;
```
Comment (jiazhang0) [inline .github/workflows/commit_skeleton_ex.yml:155]: use -f

**Review by jiazhang0 — APPROVED** [review]

**Pull request merged**